src/DynamicContent/Strings.php

```diff
@@ -28,41 +34,112 @@ class Strings {
 	 */
 	public static function filter( array $strings, $nodeId, array $element ) {
 
-		$dynamicFields = wpml_collect( isset( $element['settings']['__dynamic__'] ) ? $element['settings']['__dynamic__'] : [] );
+		$dynamicFields = self::getDynamicFields( $element );
 
-		$getStringKey = function ( WPML_PB_String $string ) {
-			return explode( '-', $string->get_name() )[0];
-		};
+		$updateFromDynamicFields = function( WPML_PB_String $string ) use ( &$dynamicFields, $nodeId ) {
+			$matchingField = $dynamicFields->first( function( Field $field, $key ) use ( $string ) {
```

Comment:
Instead of looking at the string key (which was uncertain), we are now looking for a field object which is matching the static string. Then we replace the static string with the strings for the dynamic field.